jwst/stpipe/tests/test_pipeline.py

```diff
@@ -60,9 +60,9 @@ class MultiplyBy2(Step):
 
     def process(self, image):
         with datamodels.ImageModel(image) as dm:
-            with datamodels.ImageModel() as dm2:
-                dm2.data = dm.data * 2
-                return dm2
+            dm2 = datamodels.ImageModel()
+            dm2.data = dm.data * 2
+            return dm2
```

Comment:
Is removing the with necessary since `dm2` is not associated with any file? Agree that your fix makes it clear that the `with` context manager is not necessary.

Reply:
With the current API, using the context manager on a `DataModel` that is being passed on is actually wrong. If a `DataModel` is to be treated as an I/O stream, then having a model be returned after leaving a context where it would be closed is simply against the API. Returning a `DataModel` implies that said `DataModel` will be accessed/operate on/modified later, so closing the `DataModel` is semantically incorrect.

Agreed that today, with the code base as per this PR, this example is works, since it has no other resources opened. However, as per the API, this is not guaranteed and could very well change in the future.

Also agreed that this does need to be worked out a bit more. JP-1568 has been created to account for such issues.